machines/backup.ts

```diff
@@ -0,0 +1,220 @@
+import {EventFrom, StateFrom, send} from 'xstate';
+import {createModel} from 'xstate/lib/model';
+import {AppServices} from '../shared/GlobalContext';
+import {StoreEvents} from './store';
+import Storage from '../shared/storage';
+import {compressData} from '../shared/cryptoutil/cryptoUtil';
+import {
+  getEndEventData,
+  getImpressionEventData,
+  getStartEventData,
+  sendEndEvent,
+  sendImpressionEvent,
+  sendStartEvent,
+} from '../shared/telemetry/TelemetryUtils';
+import {TelemetryConstants} from '../shared/telemetry/TelemetryConstants';
+
+const model = createModel(
+  {
+    serviceRefs: {} as AppServices,
+    dataFromStorage: {},
+    fileName: '',
+  },
+  {
+    events: {
+      DATA_BACKUP: () => ({}),
+      OK: () => ({}),
+      FETCH_DATA: () => ({}),
+      DISMISS: () => ({}),
+      STORE_RESPONSE: (response: unknown) => ({response}),
+      FILE_NAME: (filename: string) => ({filename}),
+    },
+  },
+);
+
+export const BackupEvents = model.events;
+
+export const backupMachine = model.createMachine(
```

Comment:
Can Xstate sort of ensure that this state machine is only spwaned once at a time?

Reply:
This is getting spwaned once like other machines.